packages/core/src/awsService/ec2/commands.ts

```diff
@@ -71,3 +61,16 @@ async function getSelection(node?: Ec2Node, filter?: instanceFilter): Promise<Ec
 export async function copyInstanceId(instanceId: string): Promise<void> {
     await copyToClipboard(instanceId, 'Id')
 }
+
+export async function getConnectionManager(
+    connectionManagers: Ec2ConnectionManagerMap,
+    selection: Ec2Selection
+): Promise<Ec2ConnectionManager> {
+    if (connectionManagers.has(selection.region)) {
+        return connectionManagers.get(selection.region)!
+    } else {
+        const newConnectionManager = new Ec2ConnectionManager(selection.region)
+        connectionManagers.set(selection.region, newConnectionManager)
```

Comment:
it's a bit subtle that this is actually adding to a global long-lived collection. This logic should live closer to that collection, possibly in a class or something. 

Also need to log a warning if the number of connections exceeds a big number, until we have a better way to have visbility and validation of these long-lived collections (which can caused hidden bugs / perf issues).